File: .NET/Microsoft.Recognizers.Text/Models/ExtendedModelResult.cs
namespace Microsoft.Recognizers.Text
{
    public class ExtendedModelResult : ModelResult
    {
        // Parameter Key
        public static readonly string ParentTextKey = "parentText";

        public ExtendedModelResult()
        {
        }

        public ExtendedModelResult(ModelResult modelResult)
        {
            Start = modelResult.Start;
            End = modelResult.End;
            TypeName = modelResult.TypeName;
            Resolution = modelResult.Resolution;
            Text = modelResult.Text;
        }

        public string ParentText { get; set; }
    }
}

File: Java/libraries/recognizers-text/src/main/java/com/microsoft/recognizers/text/ExtendedModelResult.java
package com.microsoft.recognizers.text;

import java.util.SortedMap;

public class ExtendedModelResult extends ModelResult {
    // Parameter Key
    public static final String ParentTextKey = "parentText";

    public final String parentText;

    public ExtendedModelResult(String text, int start, int end, String typeName, SortedMap<String, Object> resolution, String parentText) {
        super(text, start, end, typeName, resolution);
        this.parentText = parentText;
    }

    public ExtendedModelResult(ModelResult result, String parentText) {
        this(result.text, result.start, result.end, result.typeName, result.resolution, parentText);
    }
}
